### PR TITLE
Fix citations and travis build

### PR DESCRIPTION
**Pull request by JamesOwers**
# Issue
Unfortunately, [`pandoc-citeproc`](https://github.com/jgm/pandoc-citeproc) is depreciated; [citeproc-hs](https://github.com/jgm/citeproc) is now installed by default and [used by pandoc](https://pandoc.org/demos.html#examples:~:text=Markdown%20citations%3A,ieee.csl%20CITATIONS%20%2Dt%20man%20%2Do%20example24c.1). This means the installation (either via travis, install.sh, or hacking through the README.md) is broken (because installing pandoc-citeproc kicks a depreciation error).

If the user proceeds without `pandoc-citeproc`, citations don't work!

# Fix
* Fix and extend travis build
  * ~tests both mac and linux now~ linux needs work to install texlive (see comments below, it's 90% done though)
* Fix install script by removing pandoc-citeproc, and
  * made separate files for mac and linux (in case they have different requirements in future)
  * `make install` accounts for the diff files and uses the correct one automatically
* Make citations work for the example generated from `source/*.md` by:
  * Change commands in Makefile to use `--citeproc` argument
  * Fix `style/template.tex` with new contents of `pandoc -D latex` relating to citations

## Other minor contributions
* Extended examples in `source/*.md` files including:
  * some advanced equations (showing how to back off to latex if needed)
  * section referencing with `pandoc-secnos`
* Tidy commands in Makefile to highlight differences and make the output format consistent
* Version control all output types `thesis.{pdf,tex,html,docx}`, such that it is obvious if changes are made (not true for docx, so there's an argument we shouldn't version control that file since it is large)
* Add a target to Makefile so the user can `make all` to generate `thesis.{pdf,tex,html,docx}` with one command
* Output pandoc logs to separate files: `pandoc.{pdf,tex,html,docx}.log
* Extended the README.md:
    * Add quickstart for mac users
    * Add high level explanation of how pandoc processes all the files
    * Add troubleshooting

### CODE REVIEW

**Comment by JamesOwers** [comment]
This is 99% there. Travis build passes for all commands bar `make pdf`, which is erroring with `LaTeX Error: File `l3backend-xetex.def' not found.z`. If this can be fixed, this will be a decent PR to merge ☺️.

**Comment by JamesOwers** [comment]
For `.travis.yml`, linux build is nearly there now too, stuck on:
```
$ make install
bash /home/travis/build/tompollard/phd_thesis_markdown/install_linux.sh
(running on Debian, switching to user mode!)
cannot setup TLPDB in /home/travis/texmf at /usr/bin/tlmgr line 5604.
Makefile:37: recipe for target 'install' failed
make: *** [install] Error 2
The command "make install" failed and exited with 2 during .
```

**Comment by JamesOwers** [comment]
Ok, that's the build fixed for mac. Paved the way for fixing for linux next PR.

**Comment by dendrondal** [comment]
This is phenomenal work! Thank you for putting the time in to make all of these improvements. There are a few other PRs (#97 and #96) that I'm also wanting to merge, but I'm going to review yours first, as I think having a more through CI process will mean less headaches down the road. This will probably take a second, so bear with me.

File: source/09_chapter_1.md [hunk 1]
```diff
@@ -37,6 +37,8 @@ For bold, add ** on either side of the text
...
+This is a brief outline of what went into each chapter, and a section which shows how to reference headers (which are labelled automatically for you). This chapter, +@sec:introduction-with-a-citation, shows how to use citations and how to reference section headers. \*@sec:literature-review-with-maths shows how use and reference equations. \*@sec:first-research-study-with-code shows how to use and reference code. \*@sec:research-containing-a-figure shows how to use, reference, and resize pdf and jpg figures. \*@sec:research-containing-a-table shows how to use and reference tables. \*@sec:final-research-study is truly revolutionary (but shows nothing functional). **[Appendix 1](#appendix-1-some-extra-stuff)** shows how to add chapters which are not numbered, and has to be referenced manually, as does **[Appendix 2](#appendix-2-some-more-extra-stuff)**. See the base [`README.md`](https://github.com/tompollard/phd_thesis_markdown/blob/master/README.md) for how References are handled - leave `*_references.md` alone, and provide it to `pandoc` last.
```
Comment (dendrondal) [inline source/09_chapter_1.md:40]: Kudos for adding this, I didn't realize we had no secnos examples in here.

File: style/template.tex [hunk 2]
```diff
@@ -152,13 +158,36 @@
 \fi
 $endif$
 
+% $if(csl-refs)$
```
Comment (dendrondal) [inline style/template.tex:161]: What is the purpose of this block?

Comment (JamesOwers) [reply style/template.tex:161]: Check out `pandoc -D latex` to see what the standard template currently is.

**Comment by JamesOwers** [comment]
> This is phenomenal work! Thank you for putting the time in to make all of these improvements. There are a few other PRs (#97 and #96) that I'm also wanting to merge, but I'm going to review yours first, as I think having a more through CI process will mean less headaches down the road. This will probably take a second, so bear with me.

As an aside - I'm v happy to help in future since I'll be using this template to write my phd actively. The linux travis build is nearly there, I could have a crack at that next.

File: style/template.tex (hunk 2 above)
Comment (dendrondal) [reply style/template.tex:161]: Huh, interesting. Whatever works, and it certainly doesn't seem to be breaking anything!

**Comment by dendrondal** [comment]
> As an aside - I'm v happy to help in future since I'll be using this template to write my phd actively. The linux travis build is nearly there, I could have a crack at that next.

Absolutely! If you want to add docker integration to the `travis.yml` also (compatible with #97, see #96 for an example), that would also be appreciated, and well within that scope. Always happy to have more contributors, and may the writing process be as painless as possible!

**Comment by JamesOwers** [comment]
> > As an aside - I'm v happy to help in future since I'll be using this template to write my phd actively. The linux travis build is nearly there, I could have a crack at that next.
> 
> Absolutely! If you want to add docker integration to the `travis.yml` also (compatible with #97, see #96 for an example), that would also be appreciated, and well within that scope. Always happy to have more contributors, and may the writing process be as painless as possible!

Yeah will do. Never used docker before so should be fun. This PR is complicated enough so I’ll wait for this to merge (fixing any changes you’d like), then raise a separate PR for Linux + docker travis improvements.

**Comment by dendrondal** [comment]
I definitely think it's a good idea to have that as a separate request. I think this is ready to merge.

**Comment by tompollard** [comment]
@JamesOwers @dendrondal thanks for all of the work on this!

**Pull request merged**